model/task_lifecycle.go

```diff
@@ -1209,25 +1213,37 @@ func getBuildStatus(buildTasks []task.Task) buildStatus {
 		return buildStatus{status: evergreen.BuildCreated, allTasksBlocked: allTasksBlocked}
 	}
 
+	var hasUnfinishedTask bool
+	var unfinishedEssentialTasks []string
 	// Check if tasks are started but not finished.
 	for _, t := range buildTasks {
 		if t.Status == evergreen.TaskStarted {
-			return buildStatus{status: evergreen.BuildStarted}
+			hasUnfinishedTask = true
 		}
 		if t.Activated && !t.Blocked() && !t.IsFinished() {
-			return buildStatus{status: evergreen.BuildStarted}
+			hasUnfinishedTask = true
 		}
+		if t.IsEssentialToFinish && !t.IsFinished() {
+			unfinishedEssentialTasks = append(unfinishedEssentialTasks, t.Id)
+		}
+	}
+	if hasUnfinishedTask {
+		return buildStatus{status: evergreen.BuildStarted, unfinishedEssentialTasks: unfinishedEssentialTasks}
 	}
 
-	// Check if all tasks are finished but have failures.
+	// Check if tasks are finished but failed.
 	for _, t := range buildTasks {
 		if evergreen.IsFailedTaskStatus(t.Status) || t.Aborted {
-			return buildStatus{status: evergreen.BuildFailed}
+			return buildStatus{status: evergreen.BuildFailed, unfinishedEssentialTasks: unfinishedEssentialTasks}
```

Comment:
If we're just using the  length of unfinishedEssentialTasks, should we just store that instead of the list of task IDs?

Reply:
Yeah, I changed it to just an int.